### PR TITLE
feat(blocks): Add shortform video block

### PR DESCRIPTION
**Pull request by Torantulino**
### Background

We are introducing a new block, `AIShortformVideoCreatorBlock`, to enable users to create short-form videos using the [revid.ai](https://www.revid.ai/) API. This enhancement allows agents to generate engaging video content programmatically, expanding the AutoGPT platform's capabilities to include multimedia content creation. By integrating video generation, users can automate content production for social media, marketing, and other creative applications.

### Example Agent to test with
[AI Shortform Video Block Test_v3.json](https://github.com/user-attachments/files/17179535/AI.Shortform.Video.Block.Test_v3.json)

### Changes 🏗️

- **Added `AIShortformVideoCreatorBlock` in `backend/blocks/ai_shortform_video_block.py`:**
  - **Functionality:**
    - Interfaces with the revid.ai API to create short-form videos based on user-provided scripts and parameters.
    - Supports customization options like aspect ratio, resolution, frame rate, generation presets, background music, voice selection, and video style.
  - **Input Fields:**
    - `api_key`: Securely input your revid.ai API key.
    - `script`: The script content for the video, with support for text-to-speech and visual prompts.
    - `ratio`, `resolution`, `frame_rate`: Video configuration settings.
    - `generation_preset`, `background_music`, `voice`, `video_style`: Options to customize the video's aesthetic and auditory elements.
  - **Output Fields:**
    - `video_url`: URL of the generated video.
    - `error`: Error message if the video creation fails.
  - **Methods Implemented:**
    - `create_webhook`: Sets up a webhook to receive video status updates.
    - `create_video`: Sends a video creation request to the revid.ai API.
    - `check_video_status`: Polls the API for the video's processing status.
    - `wait_for_video`: Waits for the video to be processed and retrieves the video URL.
- **Updated `backend/util/settings.py`:**
  - Added `revid_api_key` to the `Secrets` class to securely store the revid.ai API key.

- **Create from scratch and execute an agent with at least 3 blocks:**
  - Created a complex demo agent that includes the new `AIShortformVideoCreatorBlock`
  - Confirmed that the agent executes successfully and the video is generated as expected.
- **Edit an agent from monitor, and confirm it executes correctly:**
  - Modified the agent's parameters, such as changing the script and background music.
  - Ran the agent and verified that the new settings are applied and the video is generated accordingly.

### CODE REVIEW

**Comment by Torantulino** [comment]
This will be a prime example of a block that would benefit from webhook support when we add it. 
This works, but in extreme cases it can be polling the API for 10 minutes.

**Comment by Torantulino** [comment]
Also this is a prime example of a block that could do with it's own documentation when we add per-block docs soon. 
Happy to be DMed to help out with documenting this one.

File: autogpt_platform/backend/backend/blocks/ai_shortform_video_block.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+        response.raise_for_status()
+        webhook_data = response.json()
+        return webhook_data["uuid"], f"https://webhook.site/{webhook_data['uuid']}"
+
+    def create_video(self, api_key: str, payload: dict) -> dict:
+        url = "https://www.revid.ai/api/public/v2/render"
+        headers = {"key": api_key}
+        response = requests.post(url, json=payload, headers=headers)
+        logger.info(f"API Response Status Code: {response.status_code}")
+        logger.info(f"API Response Content: {response.text}")
```
Comment (majdyz) [inline autogpt_platform/backend/backend/blocks/ai_shortform_video_block.py]: nit: merge these two logs ?

**Review by majdyz — DISMISSED** [review]

File: autogpt_platform/backend/backend/blocks/ai_shortform_video_block.py (hunk 1 above)
Comment (Pwuts) [reply autogpt_platform/backend/backend/blocks/ai_shortform_video_block.py]: also: these log statements are non-specific making them practically noise in production, and considering their content they should be `debug` not `info`

**Review by aarushik93 — APPROVED** [review]
Other than the comment about logs, lgtm!

File: autogpt_platform/backend/backend/blocks/ai_shortform_video_block.py (hunk 1 above)
Comment (Torantulino) [reply autogpt_platform/backend/backend/blocks/ai_shortform_video_block.py]: Done

**Review by Pwuts — DISMISSED** [review]

**Review by aarushik93 — APPROVED** [review]

**Pull request merged**